master.js

```diff
@@ -118,26 +117,25 @@ class AlAzureMaster {
         this._resourceGroup = resourceGroup ? resourceGroup : process.env.APP_RESOURCE_GROUP;
         this._webAppName = webAppName ? webAppName : process.env.WEBSITE_SITE_NAME;
         
-        // Init Azure SDK
-        if (process.env.MSI_ENDPOINT && process.env.MSI_SECRET) {
-            this._azureCreds = new msRestAzure.MSIAppServiceTokenCredentials();
-        } else {
-            const tokenCache = new fileTokenCache(m_util.getADCacheFilename(
-                'https://management.azure.com',
-                this._clientId,
-                this._domain));
-            this._azureCreds = new msRestAzure.ApplicationTokenCredentials(
-                this._clientId,
-                this._domain,
-                this._clientSecret,
-                { 'tokenCache': tokenCache });
-        }
-        this._azureWebsiteClient = new azureArmWebsite(this._azureCreds, this._subscriptionId);
         this._appStats = new AzureWebAppStats(collectorAzureFunNames);
         this._collectionStats = new AzureCollectionStats(azureContext, {outputQueueBinding: OutputStatsBinding});
         this._alAzureDlBlob = new AlAzureDlBlob(azureContext, null);
+
+        //Initialize new SDK
+        if (process.env.MSI_ENDPOINT && process.env.MSI_SECRET) {
+            const options = {
+                msiEndpoint: process.env.MSI_ENDPOINT,
+                msiSecret: process.env.MSI_SECRET,
+            };
+            this._azureCreds = new MSIAppServiceTokenCredentials(options);
+        } else {
+            this._azureCreds = new ApplicationTokenCredentials(this._clientId, this._clientSecret, this._domain);
```

Comment:
Is auth token cached now?

Reply:
It uses the default memory cache in the Token Credential object.
https://github.com/Azure/ms-rest-nodeauth/blob/adea8415de97a8076006fbd9948123d09dfdd78d/lib/credentials/tokenCredentialsBase.ts#L18